pandas/core/series.py

```diff
@@ -2588,7 +2589,14 @@ def combine(self, other, func, fill_value=None):
         elif is_extension_array_dtype(self.values):
             # The function can return something of any type, so check
             # if the type is compatible with the calling EA.
-            new_values = try_cast_to_ea(self._values, new_values)
+            if (
```

Comment:
pls don't do complicated logic here, instead do so inside ``try_cast_to_ea``, if *really* necessary

Reply:
Okay, Found out that we can't put those conditions in that file because for that we will have to import SparseArray there and that is creating the import errors.
Is there any other plan or should I replace that code in combine function?

Reply:
Or another way is that we can put the condition of the `SparseArray` in the `combine` function and other conditions in `try_cast_to_ea`. Any suggestions @jreback ?